modes/ripgrep/evil-collection-ripgrep.el

```diff
@@ -0,0 +1,45 @@
+;;; evil-collection-ripgrep.el --- Bindings for `ripgrep' -*- lexical-binding: t -*-
+
+;; Copyright (C) 2018 James Nguyen
+
+;; Author: Sid Kasivajhula <sid@countvajhula.com>
+;; Maintainer: James Nguyen <james@jojojames.com>
+;; Pierre Neidhardt <mail@ambrevar.xyz>
+;; URL: https://github.com/emacs-evil/evil-collection
+;; Version: 0.0.1
+;; Package-Requires: ((emacs "25.1"))
+;; Keywords: evil, emacs, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; Bindings for `ripgrep'.
+
+;;; Code:
+(require 'evil-collection)
+(require 'ripgrep nil t)
+
+(defconst evil-collection-ripgrep-maps '(ripgrep-search-mode-map))
+
+;;;###autoload
+(defun evil-collection-ripgrep-setup ()
+  "Set up `evil' bindings for `ripgrep'."
+  (evil-collection-define-key 'normal 'ripgrep-search-mode-map
+    "n" 'evil-search-next
+    "\C-j" 'next-error-no-select
+    "\C-k" 'previous-error-no-select))
+
+
```

Comment:
Looks like these are the bindings.

``` (define-key map "p" 'compilation-previous-error)
    (define-key map "n" 'compilation-next-error)
    (define-key map "{" 'compilation-previous-file)
    (define-key map "}" 'compilation-next-file)
    (define-key map "k" 'ripgrep/kill-buffer)
    (define-key map "C-c C-p" 'wgrep-change-to-wgrep-mode)```

Are these just not bound in this compatibility layer?

Would like to see ```wgrep-change-to-wgrep-mode``` on 'i'.

Reply:
These bindings are present in the projectile wrapped version too.
I added `{` and `}` for file navigation, and `wgrep-change-to-wgrep-mode` on `i`.
`compilation-next-error` and `compilation-previous-error` don't seem to do very much, essentially equivalent to `j` and `k`.
Did you have something in mind for `ripgrep/kill-buffer`?

Reply:
Maybe 'q' if that's the right one to kill the buffer/similar to quit-window.

Reply:
`q` does `quit-window` while in ripgrep mode. Looking at the value of `ripgrep-search-mode-map`, I see this:
```
...
 (normal-state keymap "Auxiliary keymap for Normal state"
			   (90 keymap
				   (81 . evil-quit)
				   (90 . quit-window))
			   (113 . quit-window)
...
```

113 is `q`.

Is this fine or do we want to replace the binding with `ripgrep/kill-buffer`?